src/pages/_app.tsx

```diff
@@ -1,11 +1,27 @@
-import { AppProps } from "next/app";
-import { ApolloClient, InMemoryCache } from "@apollo/client";
+import { ApolloClient, concat, HttpLink, InMemoryCache } from "@apollo/client";
+import { setContext } from "@apollo/client/link/context";
 import { ApolloProvider } from "@apollo/client/react";
-import { AuthProvider } from "../../utils/firebase/auth";
+import { AppProps } from "next/app";
 import "tailwindcss/tailwind.css";
+import { AuthProvider } from "../../utils/firebase/auth";
+import firebase from "../../utils/firebase/firebase";
+
+const httpLink = new HttpLink({ uri: process.env.NEXT_PUBLIC_API_URL });
+
+const authLink = setContext(async (_, { headers }) => {
+  const token = await firebase
```

Comment:
Do we want to consider caching the token like [this page](https://www.apollographql.com/docs/react/api/link/apollo-link-context/#caching-lookups) shows?

Reply:
No, because refreshing the token is Firebase's responsibility. With `forceRefresh` set to false, the token should be automatically cached.